### PR TITLE
cli creates and distributes staking certs now

### PR DESCRIPTION
**Pull request by arturrez**

### CODE REVIEW

File: pkg/utils/staking.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+// See the file LICENSE for licensing terms.
+package utils
+
+import (
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/staking"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+)
+
+func NewStakingCertAndKeyBytes() ([]byte, []byte, error) {
```
Comment (felipemadero) [inline pkg/utils/staking.go:11]: no need for this function, just directly use staking.NewCertAndKeyBytes on the caller

File: pkg/utils/staking.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+func NewBlsSignerCertAndKeyBytes() ([]byte, []byte, error) {
+	blsSignerKey, err := bls.NewSecretKey()
+	if err != nil {
+		return nil, nil, err
+	}
+	blsPublicBytes := bls.PublicKeyToBytes(bls.PublicFromSecretKey(blsSignerKey))
+	return bls.SecretKeyToBytes(blsSignerKey), blsPublicBytes, nil
+}
+
+func GetNodeID(certBytes []byte, keyBytes []byte) (string, error) {
```
Comment (felipemadero) [inline pkg/utils/staking.go:28]: probably best to call it ToNodeID.

File: pkg/utils/staking.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+	cert, err := staking.LoadTLSCertFromBytes(keyBytes, certBytes)
+	if err != nil {
+		return "", err
+	}
+	return ids.NodeIDFromCert(staking.CertificateFromX509(cert.Leaf)).String(), nil
```
Comment (felipemadero) [inline pkg/utils/staking.go:33]: return the nodeID, not the string, that is more general

File: cmd/nodecmd/create.go [hunk 4]
```diff
@@ -286,6 +292,48 @@ func setupBuildEnv(inventoryPath, ansibleHostIDs string) error {
 	return ansible.RunAnsiblePlaybookSetupBuildEnv(app.GetAnsibleDir(), inventoryPath, ansibleTargetHosts)
 }
 
+func GenerateStakingCertAndKey(nodesDirPath string) error {
+	certBytes, keyBytes, err := utils.NewStakingCertAndKeyBytes()
+	if err != nil {
+		return err
+	}
+	if err := os.WriteFile(filepath.Join(nodesDirPath, constants.StakerCertFileName), certBytes, 0o600); err != nil {
```
Comment (felipemadero) [inline cmd/nodecmd/create.go:300]: use a new constant for 0o600 we have another one for 0o755 I believe follow that naming

File: cmd/nodecmd/create.go [hunk 5]
```diff
@@ -286,6 +292,48 @@ func setupBuildEnv(inventoryPath, ansibleHostIDs string) error {
 	return ansible.RunAnsiblePlaybookSetupBuildEnv(app.GetAnsibleDir(), inventoryPath, ansibleTargetHosts)
 }
 
+func GenerateStakingCertAndKey(nodesDirPath string) error {
```
[the remark this answers is not in the document]
Comment (felipemadero) [reply cmd/nodecmd/create.go:295]: FIX call it generateNodeCertAndKeys

File: cmd/nodecmd/create.go [hunk 6]
```diff
@@ -286,6 +292,48 @@ func setupBuildEnv(inventoryPath, ansibleHostIDs string) error {
...
+		return err
+	}
+	if err := os.WriteFile(filepath.Join(nodesDirPath, constants.BLSFileName), blsSignerKeyBytes, 0o600); err != nil {
+		return err
+	}
+	return nil
+}
+
+func DistributeStakingCertAndKey(ansibleHostIDs []string, inventoryPath string) error {
+	ux.Logger.PrintToUser("Copying %s and %s staker.key to remote machine(s)...", constants.StakerCertFileName, constants.StakerKeyFileName)
```
[the remark this answers is not in the document]
Comment (felipemadero) [reply cmd/nodecmd/create.go:317]: include all 3 kinds of files in msgss

File: pkg/ansible/playbook/copyStakingFiles.yml [hunk 7]
```diff
@@ -5,18 +5,20 @@
...
-        src: /home/ubuntu/.avalanchego/staking/signer.key
-        dest: "{{ nodesDirPath }}/{{ host_id.stdout }}/"
-        flat: true
+    - name: ensures {{ project_root }}/conf dir exists
+      file: 
+        path: "/home/ubuntu/.avalanchego/staking/"
+        state: directory
+        recurse: yes
+    - name: copy staker.crt to remote machine
+      copy:
```
[the remark this answers is not in the document]
Comment (arturrez) [reply pkg/ansible/playbook/copyStakingFiles.yml:14]: I would rather not address this as ansible is about to be deprocated

File: cmd/nodecmd/create.go [hunk 8]
```diff
@@ -286,6 +292,48 @@ func setupBuildEnv(inventoryPath, ansibleHostIDs string) error {
...
+	if err != nil {
+		return err
+	}
+	if err := os.WriteFile(filepath.Join(nodesDirPath, constants.BLSFileName), blsSignerKeyBytes, 0o600); err != nil {
+		return err
+	}
+	return nil
+}
+
+func DistributeStakingCertAndKey(ansibleHostIDs []string, inventoryPath string) error {
```
[the remark this answers is not in the document]
Comment (arturrez) [reply cmd/nodecmd/create.go:316]: wdym?

File: pkg/ansible/playbook/copyStakingFiles.yml (hunk 7 above)
[the remark this answers is not in the document]
Comment (arturrez) [reply pkg/ansible/playbook/copyStakingFiles.yml:14]: but I think we can :)

**Review by felipemadero — APPROVED** [review]

**Pull request merged**